scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,246 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+arg_names=""
+arg_kubeconfig=""
+arg_context=""
+arg_ip_addr=""
+arg_merge=false
+arg_output=""
+arg_verbose=true
+
+
+display_help() {
+  cat << EOF
+Usage: $0 [--kubeconfig <filename>] [--context <name>] [--replace-local-ip-address <address>] [--merge] [-o <filename>] [-V] [-X]
+--kubeconfig <filename>                 use the specified kubeconfig
+--context <name>                        use the specified context
+--names <name1>,<name2>                 comma separated list of KubeFlex Control Planes names to import, instead of default *all*
+--replace-local-ip-address <address>    replaces server addresses "127.0.0.1" with <address>
+--merge                                 merge the control planes contexts with the existing cluster contexts
+-o <filename>                           specify a different kubeconfig file to save the contexts (- for stdout)
+--silent                                no information output
+-X                                      enable verbose execution of the script for debugging
+EOF
+}
+
+
+get_kubeconfig() {
+    context="$1"
+    cp_name="$2"
+    cp_type="$3"
+    ip_addr="$4"
+
+    echov "Getting the kubeconfig of Control Plane \"$cp_name\" of type \"$cp_type\" in context \"$context\":"
+
+    # wait for CP ready
+    while [[ $(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do
+        echov "* Waiting for Control Plane \"$cp_name\" in context \"$context\" to be Ready..."
+        sleep 5
+    done
+
+    # get out of cluster kubeconfig
+    if [[ "$cp_type" == "host" ]] ; then
+        echov "- Using cluster context \"${context}\""
+        kubeconfig=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context config view --flatten --minify)
+    else
+        # determine the secret name and namespace
+                    key=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.key}')
+            secret_name=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.name}')
+        secret_namespace=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.namespace}')
+        # get the kubeconfig
+        echov "- Using \"$key\" from \"$secret_name\" secret in \"$secret_namespace\""
+        kubeconfig=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}" | base64 -d)
+    fi
+
+    # make kubeconfig unique for the control plane
+    echov "- Making the kubeconfig unique..."
+    cluster=$(KUBECONFIG="$in_KUBECONFIG" kubectl --kubeconfig <(echo "$kubeconfig") config get-clusters | tail -n +2)
```

Comment:
The kubectl command here is being given a kubeconfig in two ways; that is confusing, surely only one is needed?

Doesn't `--kubeconfig <(nested command)` put the output of `nested command` on the outer command line, following `--kubeconfig`? This would follow `--kubeconfig` with kubeconfig _contents_ rather than a filename, right?